src/components/edit-profile/EditProfileModal.jsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import Modal from '@mui/material/Modal';
+import './EditProfileModal.css'
+import { editUserProfile } from '../../reducers/authSlice';
+import { TextField } from '@mui/material';
+import { useDispatch } from 'react-redux';
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  boxShadow: 24,
+  p: 4,
+  borderRadius:"5px"
+};
+
+export function EditProfileModal({profileUser}) {
+  const [open, setOpen] = React.useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  const dispatch = useDispatch()
+const [editProfile, setEditProfile] = React.useState({...profileUser})
```

Comment:
If profileUser is a state in parent component instead of again creating new State as editProfile you can pass the setter method setProfileUSer. and use that in this component

Reply:
okay, i will make suggested changes in the next PR